packages/sdk/src/transactions/PSBTBuilder.ts

```diff
@@ -0,0 +1,468 @@
+import { MINIMUM_AMOUNT_IN_SATS } from "../constants";
+import { FeeEstimator } from "../fee/FeeEstimator";
+import type { InputsToSign } from "../inscription/types";
+import { BaseDatasource } from "../modules/BaseDatasource";
+import { JsonRpcDatasource } from "../modules/JsonRpcDatasource";
+import {
+  convertSatoshisToBTC,
+  generateTxUniqueIdentifier,
+  getNetwork,
+  toXOnly,
+} from "../utils";
+import { InputType, processInput } from "./helper";
+import type { Output, UTXOLimited } from "./types";
+
+export interface PSBTBuilderOptions {
+  address: string;
+  changeAddress?: string;
+  feeRate: number;
+  network: Network;
+  outputs: Output[];
+  publicKey: string;
+  autoAdjustment?: boolean;
+  instantTradeMode?: boolean;
+  datasource?: BaseDatasource;
+}
+
+export type InjectableInput = {
+  injectionIndex: number;
+  txInput: unknown;
+  sats: number;
+  standardInput: InputType;
+};
+
+export interface InjectableOutput {
+  injectionIndex: number;
+  txOutput: unknown;
+  sats: number;
+  standardOutput: unknown;
+}
+
+export class PSBTBuilder extends FeeEstimator {
+  protected address: string;
+
+  protected changeAddress?: string;
+
+  protected changeAmount = 0;
+
+  protected datasource: BaseDatasource;
+
+  protected injectableInputs: InjectableInput[] = [];
+
+  protected injectableOutputs: InjectableOutput[] = [];
+
+  protected inputAmount = 0;
+
+  protected inputs: InputType[] = [];
+
+  protected outputAmount = 0;
+
+  protected outputs: Output[] = [];
+
+  protected psbt: Psbt;
+
+  protected publicKey: string;
+
+  protected rbf = true;
+
+  protected utxos: UTXOLimited[] = [];
+
+  protected usedUTXOs: string[] = [];
+
+  private autoAdjustment: boolean;
+
+  private instantTradeMode: boolean;
+
+  private nativeNetwork: networks.Network;
+
+  private noMoreUTXOS = false;
+
+  get data() {
+    return {
+      fee: this.fee,
+      virtualSize: this.virtualSize,
+      weight: this.weight,
+      changeAmount: this.changeAmount,
+      inputAmount: this.inputAmount,
+      outputAmount: this.outputAmount,
+    };
+  }
+
+  constructor({
+    address,
+    changeAddress,
+    datasource,
+    feeRate,
+    network,
+    publicKey,
+    outputs,
+    autoAdjustment = true,
+    instantTradeMode = false,
+  }: PSBTBuilderOptions) {
+    super({
+      feeRate,
+      network,
+    });
+    this.address = address;
+    this.changeAddress = changeAddress;
+    this.datasource =
+      datasource || new JsonRpcDatasource({ network: this.network });
+    this.outputs = outputs;
+    this.nativeNetwork = getNetwork(network);
+    this.publicKey = publicKey;
+
+    this.autoAdjustment = autoAdjustment;
+    this.instantTradeMode = instantTradeMode;
+
+    this.psbt = new Psbt({ network: this.nativeNetwork });
+  }
+
+  toPSBT() {
+    return this.psbt;
+  }
+
+  toHex() {
+    return this.psbt.toHex();
+  }
+
+  toBase64() {
+    return this.psbt.toBase64();
+  }
+
+  enableRBF() {
```

Comment:
Can we combine this into one method? Maybe toggleRBF? Just need to pass the value

Reply:
`setRBF(value)` works?